test/typescript/ts.ts

```diff
@@ -45,6 +45,27 @@ test("should work in typescript", t => {
     t.end();
 });
 
+test("typescript class with constructor params", t => {
```

Comment:
So it wouldn't have to parse the function each time? That makes sense, I can look at that. I was hesitant to change more than absolutely necessary to minimize breaking something. :)